lib/WP_Auth0_LoginManager.php

```diff
@@ -231,34 +231,44 @@ public function redirect_login() {
     if ( empty( $data->access_token ) ) {
 
       // Look for clues as to what went wrong
-      throw new WP_Auth0_LoginFlowValidationException(
-        ! empty( $data->error_description ) ? $data->error_description : __( 'Unknown error', 'wp-auth0' ),
-        ( ! empty( $data->error ) ? $data->error : $exchange_resp_code )
-      );
+      $e_message = ! empty( $data->error_description ) ? $data->error_description : __( 'Unknown error', 'wp-auth0' );
+      $e_code = ! empty( $data->error ) ? $data->error : $exchange_resp_code;
+      throw new WP_Auth0_LoginFlowValidationException( $e_message, $e_code );
     }
 
+    // Decode our incoming ID token for the Auth0 user_id
     $decoded_token = JWT::decode(
       $data->id_token,
       $this->a0_options->get_client_secret_as_key(),
       array( $this->a0_options->get_client_signing_algorithm() )
     );
 
-    $userinfo_resp = WP_Auth0_Api_Client::get_user(
-      $this->a0_options->get( 'domain' ),
-      WP_Auth0_Api_Client::get_client_token(),
-      $decoded_token->sub
-    );
+    // Attempt to authenticate with the Management API
+    $client_credentials_token = WP_Auth0_Api_Client::get_client_token();
+    $userinfo_resp = null;
+
+    if ( $client_credentials_token ) {
+
+        $userinfo_resp = WP_Auth0_Api_Client::get_user(
+            $this->a0_options->get( 'domain' ),
+            $client_credentials_token,
+            $decoded_token->sub
+        );
+    }
 
     $userinfo_resp_code = (int) wp_remote_retrieve_response_code( $userinfo_resp );
     $userinfo_resp_body = wp_remote_retrieve_body( $userinfo_resp );
 
-    if ( $userinfo_resp instanceof WP_Error || 200 !== $userinfo_resp_code || empty( $userinfo_resp_body ) ) {
+    // Management API call failed
+    if ( 200 !== $userinfo_resp_code || empty( $userinfo_resp_body ) ) {
```

Comment:
What I meant is remove this 2 conditions and only keep the `$userinfo_resp instanceof WP_Error` one. Why did you choose to remove the other?

Reply:
`$userinfo_resp` is only a `WP_Error` if something went wrong during the POST call. If it succeeds but you get a 401 or 404, the response code is what needs to be checked.